src/routes/users/fetch-users.e2e.test.js

```diff
@@ -0,0 +1,59 @@
+const { expect } = require('chai')
+const { fixtures, setupDB } = require('../../../test/utils')
+const fetchUsers = require('../../services/users/fetch-users')
+
+describe('GET /users', () => {    
+	describe('Success', () => {
+		let adminUser
+		let accessToken
+		beforeEach(async () =>{
+			await setupDB()
+			await fixtures.insertUser()
+			adminUser = await fixtures.insertUser({
+				role : 'admin'
+			}) 
+			const body = { id: adminUser.id, email: adminUser.email }
+			accessToken = fixtures.generateAccessToken(
+				{user : body}
+			)
+		})
+		it('should return 200 if users are fetched', async () => 
+			fixtures.api()
+				.get(`/api/v1/users`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.expect(200)
+		)
+		it('should return the right number of users', async() =>{
+			const actualUsers = await fetchUsers()
```

Comment:
So the thing here is that, the `GET /users` endpoint uses this `fetchUsers` service so we can't use that same service function in the tests else we'll not know when it breaks

in testing this endpoint, we're trying to confirm the api along with the service(fetchUsers) is working as expected

Reply:
thanks so much for the explanation. I get you perfectly